nucleus/admin/cli/src/main/java/com/sun/enterprise/admin/cli/CLICommand.java

```diff
@@ -1164,8 +1164,8 @@ protected String getPassword(ParamModel opt, String defaultPassword,
             confirmationPrompt = strings.get("NewPasswordConfirmationPrompt", passwordName);
         }
 
-        char[] newpasswordAgain = readPassword(confirmationPrompt);
-        if (!Arrays.equals(newpassword, newpasswordAgain)) {
+        String newpasswordAgain = readPassword(confirmationPrompt);
+        if (newpassword.equals(newpasswordAgain)) {
```

Comment:
logic looks flipped. Missing a NOT (!) probably.

Reply:
Good spot sir!